### PR TITLE
Build routes even less

### PR DESCRIPTION
**Pull request by Hixie**

### CODE REVIEW

**Comment by Hixie** [comment]
cc @chunhtai

File: packages/flutter/test/widgets/navigator_test.dart [hunk 1]
```diff
@@ -1141,7 +1141,50 @@ void main() {
...
-    expect(log, <String>['building B', 'building C', 'found C', 'building D', 'building C', 'found C']);
+    expect(log, <String>['building B', 'building C', 'found C', 'building D']);
+  });
+
+  testWidgets('Routes don\'t rebuild just because their animations ended', (WidgetTester tester) async {
+    final GlobalKey<NavigatorState> key = GlobalKey<NavigatorState>();
+    final List<String> log = <String>[];
+    Route<dynamic> nextRoute = PageRouteBuilder<int>(
+      pageBuilder: (BuildContext context, Animation<double> animation, Animation<double> secondaryAnimation) {
+        log.add('building page 1 - ${ModalRoute.of(context).canPop}');
```
Comment (chunhtai) [inline packages/flutter/test/widgets/navigator_test.dart:1152]: Why do we care about can pop?

**Review by chunhtai — COMMENTED** [review]
Thanks for the optimization!

File: packages/flutter/test/widgets/navigator_test.dart (hunk 1 above)
Comment (Hixie) [reply packages/flutter/test/widgets/navigator_test.dart:1152]: It used to change value (that's why the pages rebuilt, basically). This is mostly making sure that it isn't rebuilding even if it's changing value.

Comment (Hixie) [reply packages/flutter/test/widgets/navigator_test.dart:1152]: Ah, I should update those docs. Good catch!

**Comment by Hixie** [comment]
Docs updated, PTAL.

**Review by chunhtai — APPROVED** [review]
LGTM

**Pull request merged**